### DISCUSSION TITLE
SWATH data to grid

### DISCUSSION THREAD

**Discussion by rich-HJ** [NCAS-CMS/cf-python#192] [Q&A] [answered]
Hi all,

I thought I'd get the discussion rolling. I am trying to encourage use of cfpython at CEH. 

One of the questions I have been asked is can this be used to interpolate/regrid satellite swath data to a grid? The swath files have long and lat as ancillary dimensions but no dimension coordinate. 

I can provide example files if this helps but we just wondered if this was a trivial method that we had not noticed.

Thanks in advance,

Rich

**Comment by davidhassell** [comment]
Hi Rich,

I have not tried cf-python with swath data before, but it can deal with some unstructured grids - those with auxiliary latitudes and longitudes but not dimension coordinate - such as tri-polar grids. 

Are the latitude and longitude coordinates 1-d or 2-d?

If you can provide a sample file, I can investigate. If it doesn't "just work", it _may_ be a relatively minor change to get it going.

Thanks,
David

**Reply by rich-HJ** [reply]
Hi David,

That was quick. 2d lon and lat. Pasting a nc file here seems to upset the system. Where would be a convenient spot to drop it?



      int lat(numRows, numCells) ;
                lat:standard_name = "grid_latitude" ;
                lat:long_name = "latitude" ;
                lat:units = "degree_north" ;
                lat:valid_min = -90000000 ;
                lat:valid_max = 90000000 ;
                lat:_FillValue = -2147483647 ;
                lat:scale_factor = 1.e-06f ;
        int lon(numRows, numCells) ;
                lon:standard_name = "grid_longitude" ;
                lon:long_name = "longitude" ;
                lon:units = "degree_east" ;
                lon:valid_min = -180000000 ;
                lon:valid_max = 180000000 ;
                lon:_FillValue = -2147483647 ;
                lon:scale_factor = 1.e-06f ;

**Comment by davidhassell** [comment, marked as answer]
Hi Rich,

On the face of it, your data should be OK with the existing cf-python, as your lats and lons are 2-d. You'd need to tell it which dimension is "X" and which is "Y", using the _src_axes_ keyword.

I don't know about attaching a netCDF file in GitHub (there must be a way, right?)

David

P.S. The `standard_name` and `units` and inconsistent!

**Reply by rich-HJ** [reply]
Hi David,

Great thanks. I'll try and help them to that solution. ... and get them to post their own queries next time. 

This page only accepts text files and images at the moment. I guess they don't want mutli-MB files uploaded here.

Odd that there are errors in the meta data. I might pass that up to the data provider.

Cheers,

Rich
